### PR TITLE
Replace FooTable with DataTables

### PR DESCRIPTION
**Pull request by kitsuta**
Fixes https://github.com/magfest/ubersystem/issues/1273 by replacing the broken FooTable plugin with DataTables.

### CODE REVIEW

**Comment by kitsuta** [comment]
Oh yeah, and since DataTables has built-in search functionality, I took out the broken search box in the Groups page. So this fixes https://github.com/magfest/ubersystem/issues/1353 as well.

**Comment by binary1230** [comment]
all this CSS stuff is a bit over my head but looks great from my perspective, nice fix +1

File: bower.json [hunk 1]
```diff
@@ -1,39 +1,39 @@
...
+    "jquery-hoverIntent": "1.8.1",
+    "select2": "4.0.0",
+    "jQuery-contextMenu": "1.6.6",
+    "jquery-confirm": "2.3.1",
+    "toastr": "2.1.1",
+    "geocomplete": "1.6.5",
+    "angular": "1.3.15",
+    "angular-route": "1.3.15",
+    "angular-bootstrap": "0.13.0",
+    "datatables": "~1.10.8"
```
Comment (EliAndrewC) [inline bower.json]: You should remove the ``~`` here.  What that means is "Use 1.10.8 or any later version in the 1.10.x line."  Which means that the next time someone adds a new dependency and runs bower, they could literally end up upgrading the version of ``datatables`` they're using without even realizing it.  This is generally undesirable, so we should pin our dependencies to the exact version whenever possible.  (Ideally we'd be using ``npm-shrinkwrap`` to also pin the dependencies of our dependencies, but that's a different ticket.)

File: bower.json [hunk 2]
```diff
@@ -1,39 +1,39 @@
...
-        "footable": "2.0.3",
-        "geocomplete": "1.6.5",
-        "angular": "1.3.15",
-        "angular-route": "1.3.15",
-        "angular-bootstrap": "0.13.0"
-    },
-    "resolutions": {
-        "jquery": ">= 1.9.0",
-        "bootstrap": "3.2.0"
-    }
```
Comment (EliAndrewC) [inline bower.json]: Dunno why the indentation changed all of a sudden, but 2 spaces is probably more standard for json files anyway, I suppose.

**Comment by EliAndrewC** [comment]
> Oh yeah, and since DataTables has built-in search functionality, I took out the broken search box in the Groups page. So this fixes #1353 as well.

The group search (when it worked) also searched on fields which are not shown in the table, such as admin notes.  So we should probably put that back in.  Although then again, we can always leave it out and see if anyone complains, which is generally a good way to tell whether a feature is actually being used :P

**Comment by binary1230** [comment]
funny story I looked back in time and the group search was never actually implemented.  after the UI rewrite, that search box was added as a placeholder and never hooked up to anything.

while that's true for Group search, attendee search has a working searchbox, and yea, searching for non-listed fields is a good call.

**Comment by thaeli** [comment]
If we are hooking up search to non-listed fields.. let's use server side
processing for the tables:
http://datatables.net/examples/data_sources/server_side.html

On Mon, Aug 24, 2015 at 3:00 PM, Dominic Cerquetti <notifications@github.com

**Comment by earl7399** [comment]
Danielle needs search functionality three weeks ago now.  Can we merge what
is working while we decide and then possibly re-tool?

--
Robert Earl <r.a.earl@gmail.com>
FAX: 1-440-919-5086

**Comment by binary1230** [comment]
oh yea sorry, thought I had +1'd, though Eli's notes are probably worth addressing.  however, probably a good thing to just go merge as-is and a separate PR or issue for the fixes.

**Pull request merged**